opentelemetry-example-app/tests/test_flask_example.py

```diff
@@ -1,14 +1,57 @@
 import unittest
+from unittest import mock
+
+import requests
+from werkzeug.test import Client
+from werkzeug.wrappers import BaseResponse
 
 import opentelemetry_example_app.flask_example as flask_example
+from opentelemetry.sdk import trace
+from opentelemetry.sdk.context.propagation import b3_format
 
 
 class TestFlaskExample(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         cls.app = flask_example.app
 
+    def setUp(self):
+        mocked_response = requests.models.Response()
+        mocked_response.status_code = 200
+        mocked_response.reason = "Roger that!"
+        self.send_patcher = mock.patch.object(
+            requests.Session,
+            "send",
+            autospec=True,
+            spec_set=True,
+            return_value=mocked_response,
+        )
+        self.send = self.send_patcher.start()
+
+    def tearDown(self):
+        self.send_patcher.stop()
+
     def test_full_path(self):
```

Comment:
this is now a full-fledged propagation test. We could probably build on this once we have stuff like exporters.

@c24t I think this will work for one of the test cases we discussed, although not as comprehensive as bringing up a full server.